assets/src/components/SiteModal.tsx

```diff
@@ -6,32 +10,57 @@ import {
 	Notice,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { useState, useMemo } from 'react';
-import { isValidUrl, withTrailingSlash } from '../js/utils';
 
 /**
- * Site Modal component for adding/editing a site.
- *
- * @param {Object}   props              - Component properties.
- * @param {Object}   props.formData     - Current form data.
- * @param {Function} props.setFormData  - Function to update form data.
- * @param {Function} props.onSubmit     - Function to call on form submission.
- * @param {Function} props.onClose      - Function to call on modal close.
- * @param {boolean}  props.editing      - Whether the modal is in editing mode.
- * @param {Object}   props.originalData - Original data for comparison when editing.
- * @return {JSX.Element} Rendered component.
+ * Internal dependencies
  */
-const SiteModal = ( { formData, setFormData, onSubmit, onClose, editing, originalData = {} } ) => {
-	const [ errors, setErrors ] = useState( {
+import { isValidUrl } from '../js/utils';
+import type { defaultBrandSite } from '@/admin/settings/page';
+
+interface SiteModalProps {
+	formData: typeof defaultBrandSite;
+	setFormData: ( data: typeof defaultBrandSite ) => void;
+	onSubmit: () => Promise< Response | void >;
+	onClose: () => void;
+	editing: boolean;
+	sites: typeof defaultBrandSite[];
+	originalData: typeof defaultBrandSite | undefined;
+}
+
+interface ErrorsType {
+	name: string;
+	url: string;
+	api_key: string;
+	message: string;
+}
+
+const SiteModal = (
+	{ formData, setFormData, onSubmit, onClose, editing, sites, originalData }
+	: SiteModalProps,
+) => {
+	const [ errors, setErrors ] = useState< ErrorsType >( {
```

Comment:
IMO if you're only using a type in a single place just inline it. No reason to add the congnitive load of jumping around a file for definitions.